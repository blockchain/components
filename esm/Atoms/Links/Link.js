function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  font-weight: ", ";\n  font-size: ", ";\n  text-transform: ", ";\n  font-style: ", ";\n  color: ", ";\n  opacity: ", ";\n  cursor: pointer;\n\n  &:hover {\n    color: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
var BaseLink = styled.a(_templateObject(), function (props) {
  return props.theme['fontPrimary'];
}, function (props) {
  return props.weight;
}, function (props) {
  return props.size;
}, function (props) {
  return props.uppercase ? 'uppercase' : props.capitalize ? 'capitalize' : 'none';
}, function (props) {
  return props.italic ? 'italic' : 'normal';
}, function (props) {
  return props.theme['linkColor'];
}, function (props) {
  return props.opacity;
}, function (props) {
  return props.theme['linkHoverColor'];
});

var Link = function Link(_ref) {
  var children = _ref.children,
      weight = _ref.weight,
      size = _ref.size,
      uppercase = _ref.uppercase,
      capitalize = _ref.capitalize,
      italic = _ref.italic,
      opacity = _ref.opacity,
      rest = _objectWithoutProperties(_ref, ["children", "weight", "size", "uppercase", "capitalize", "italic", "opacity"]);

  return React.createElement(BaseLink, _extends({
    capitalize: capitalize,
    italic: italic,
    opacity: opacity,
    size: size,
    uppercase: uppercase,
    weight: weight
  }, rest), children);
};

Link.propTypes = {
  capitalize: PropTypes.bool,
  children: PropTypes.node,
  italic: PropTypes.bool,
  opacity: PropTypes.number,
  size: PropTypes.string,
  uppercase: PropTypes.bool,
  weight: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700])
};
Link.defaultProps = {
  capitalize: false,
  italic: false,
  opacity: 1,
  size: '14px',
  uppercase: false,
  weight: 400
};
export default Link;