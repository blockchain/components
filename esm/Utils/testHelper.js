import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}