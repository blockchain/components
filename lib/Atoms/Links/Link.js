"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  font-weight: ", ";\n  font-size: ", ";\n  text-transform: ", ";\n  font-style: ", ";\n  color: ", ";\n  opacity: ", ";\n  cursor: pointer;\n\n  &:hover {\n    color: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var BaseLink = _styledComponents.default.a(_templateObject(), function (props) {
  return props.theme['fontPrimary'];
}, function (props) {
  return props.weight;
}, function (props) {
  return props.size;
}, function (props) {
  return props.uppercase ? 'uppercase' : props.capitalize ? 'capitalize' : 'none';
}, function (props) {
  return props.italic ? 'italic' : 'normal';
}, function (props) {
  return props.theme['linkColor'];
}, function (props) {
  return props.opacity;
}, function (props) {
  return props.theme['linkHoverColor'];
});

var Link = function Link(_ref) {
  var children = _ref.children,
      weight = _ref.weight,
      size = _ref.size,
      uppercase = _ref.uppercase,
      capitalize = _ref.capitalize,
      italic = _ref.italic,
      opacity = _ref.opacity,
      rest = _objectWithoutProperties(_ref, ["children", "weight", "size", "uppercase", "capitalize", "italic", "opacity"]);

  return _react.default.createElement(BaseLink, _extends({
    capitalize: capitalize,
    italic: italic,
    opacity: opacity,
    size: size,
    uppercase: uppercase,
    weight: weight
  }, rest), children);
};

Link.propTypes = {
  capitalize: _propTypes.default.bool,
  children: _propTypes.default.node,
  italic: _propTypes.default.bool,
  opacity: _propTypes.default.number,
  size: _propTypes.default.string,
  uppercase: _propTypes.default.bool,
  weight: _propTypes.default.oneOf([100, 200, 300, 400, 500, 600, 700])
};
Link.defaultProps = {
  capitalize: false,
  italic: false,
  opacity: 1,
  size: '14px',
  uppercase: false,
  weight: 400
};
var _default = Link;
exports.default = _default;