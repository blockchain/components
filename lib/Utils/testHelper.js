"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}