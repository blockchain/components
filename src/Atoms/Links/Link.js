// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

import injectE2E from '../../Tools/injectE2E'

type PropsType = {
  +capitalize?: boolean,
  +children?: React.Node,
  +e2e?: string,
  +italic?: boolean,
  +opacity?: number,
  +size?: string,
  +uppercase?: boolean,
  +weight?: 100 | 200 | 300 | 400 | 500 | 600 | 700,
}

const Wrapper = styled.a.attrs({
  'data-e2e': injectE2E,
})`
  font-family: ${(props) => props.theme['fontPrimary']};
  font-weight: ${(props) => props.weight};
  font-size: ${(props) => props.size};
  text-transform: ${(props) =>
    props.uppercase ? 'uppercase' : props.capitalize ? 'capitalize' : 'none'};
  font-style: ${(props) => (props.italic ? 'italic' : 'normal')};
  color: ${(props) => props.theme['linkColor']};
  opacity: ${(props) => props.opacity};
  cursor: pointer;
  text-decoration: none;

  &:hover {
    color: ${(props) => props.theme['linkHoverColor']};
    text-decoration: none;
  }
`

const Link = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

Link.propTypes = {
  capitalize: PropTypes.bool,
  children: PropTypes.node,
  e2e: PropTypes.string,
  italic: PropTypes.bool,
  opacity: PropTypes.number,
  size: PropTypes.string,
  uppercase: PropTypes.bool,
  weight: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700]),
}

Link.defaultProps = {
  capitalize: false,
  e2e: 'defaultLink',
  italic: false,
  opacity: 1,
  size: '1rem',
  uppercase: false,
  weight: 400,
}

export default Link
