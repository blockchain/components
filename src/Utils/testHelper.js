import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
