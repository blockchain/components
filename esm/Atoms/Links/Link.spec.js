import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import Link from './Link'

describe('Link', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<Link>MyLink</Link>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with color', () => {
    const tree = mountWithTheme(<Link color="gold">MyLink</Link>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with size', () => {
    const tree = mountWithTheme(<Link size="32px">MyLink</Link>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with weight', () => {
    const tree = mountWithTheme(<Link weight={500}>MyLink</Link>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with uppercase', () => {
    const tree = mountWithTheme(<Link uppercase>MyLink</Link>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with capitalize', () => {
    const tree = mountWithTheme(<Link capitalize>MyLink</Link>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with italic', () => {
    const tree = mountWithTheme(<Link italic>MyLink</Link>)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with opacity', () => {
    const tree = mountWithTheme(<Link opacity={0.5}>MyLink</Link>)

    expect(tree).toMatchSnapshot()
  })
})
