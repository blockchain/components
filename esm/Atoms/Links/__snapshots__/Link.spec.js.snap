// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 14px;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  color: #004A7C;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Link
    capitalize={false}
    italic={false}
    opacity={1}
    size="14px"
    uppercase={false}
    weight={400}
  >
    <styled.a
      capitalize={false}
      italic={false}
      opacity={1}
      size="14px"
      uppercase={false}
      weight={400}
    >
      <a
        className="c0"
        opacity={1}
        size="14px"
      >
        MyLink
      </a>
    </styled.a>
  </Link>
</ThemeProvider>
`;

exports[`Link should render correctly with capitalize 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 14px;
  text-transform: capitalize;
  font-style: normal;
  color: #545456;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  color: #004A7C;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Link
    capitalize={true}
    italic={false}
    opacity={1}
    size="14px"
    uppercase={false}
    weight={400}
  >
    <styled.a
      capitalize={true}
      italic={false}
      opacity={1}
      size="14px"
      uppercase={false}
      weight={400}
    >
      <a
        className="c0"
        opacity={1}
        size="14px"
      >
        MyLink
      </a>
    </styled.a>
  </Link>
</ThemeProvider>
`;

exports[`Link should render correctly with italic 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 14px;
  text-transform: none;
  font-style: italic;
  color: #545456;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  color: #004A7C;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Link
    capitalize={false}
    italic={true}
    opacity={1}
    size="14px"
    uppercase={false}
    weight={400}
  >
    <styled.a
      capitalize={false}
      italic={true}
      opacity={1}
      size="14px"
      uppercase={false}
      weight={400}
    >
      <a
        className="c0"
        opacity={1}
        size="14px"
      >
        MyLink
      </a>
    </styled.a>
  </Link>
</ThemeProvider>
`;

exports[`Link should render correctly with opacity 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 14px;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 0.5;
  cursor: pointer;
}

.c0:hover {
  color: #004A7C;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Link
    capitalize={false}
    italic={false}
    opacity={0.5}
    size="14px"
    uppercase={false}
    weight={400}
  >
    <styled.a
      capitalize={false}
      italic={false}
      opacity={0.5}
      size="14px"
      uppercase={false}
      weight={400}
    >
      <a
        className="c0"
        opacity={0.5}
        size="14px"
      >
        MyLink
      </a>
    </styled.a>
  </Link>
</ThemeProvider>
`;

exports[`Link should render correctly with size 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 32px;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  color: #004A7C;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Link
    capitalize={false}
    italic={false}
    opacity={1}
    size="32px"
    uppercase={false}
    weight={400}
  >
    <styled.a
      capitalize={false}
      italic={false}
      opacity={1}
      size="32px"
      uppercase={false}
      weight={400}
    >
      <a
        className="c0"
        opacity={1}
        size="32px"
      >
        MyLink
      </a>
    </styled.a>
  </Link>
</ThemeProvider>
`;

exports[`Link should render correctly with uppercase 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 14px;
  text-transform: uppercase;
  font-style: normal;
  color: #545456;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  color: #004A7C;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Link
    capitalize={false}
    italic={false}
    opacity={1}
    size="14px"
    uppercase={true}
    weight={400}
  >
    <styled.a
      capitalize={false}
      italic={false}
      opacity={1}
      size="14px"
      uppercase={true}
      weight={400}
    >
      <a
        className="c0"
        opacity={1}
        size="14px"
      >
        MyLink
      </a>
    </styled.a>
  </Link>
</ThemeProvider>
`;

exports[`Link should render correctly with weight 1`] = `
.c0 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  color: #004A7C;
}

<ThemeProvider
  theme={
    Object {
      "buttonBackgroundColor": "#004A7C",
      "buttonColor": "#FFF",
      "buttonHoverColor": "#0068af",
      "cartridgeBackgroundColor": "#004A7C",
      "cartridgeColor": "#FFF",
      "circleBadgeBackgroundColor": "#CCC",
      "circleBadgeHoverColor": "#004A7C",
      "dropdownBackgroundColor": "rgba(0, 0, 0, 0)",
      "dropdownColor": "#545456",
      "fontPrimary": "'IBM Plex Sans', 'Helvetica Neue', Arial, sans-serif;",
      "iconColor": "#545456",
      "iconHoverColor": "#757679",
      "inputBackgroundColor": "#FFF",
      "inputBorderColor": "#CCC",
      "inputColor": "#545456",
      "linkColor": "#545456",
      "linkHoverColor": "#004A7C",
      "panelBackgroundColor": "#FFF",
      "panelShadowColor": "#EEF2F5",
      "searchBarIconBackgroundColor": "#FFCA00",
      "searchBarInputBackgroundColor": "#FFF",
      "searchBarUnderlinedBackgroundColor": "rgba(0, 0, 0, 0)",
      "searchBarUnderlinedBorderColor": "#CCC",
      "sectionBackgroundColor": "#004A7C",
      "tabMenuBarColor": "#FFCA00",
      "tabMenuBorderColor": "#EEF2F5",
      "tableCellEvenBackgroundColor": "#F8F8F8",
      "tableCellOddBackgroundColor": "#FFF",
      "tableHeaderColor": "#999B9E",
      "tableShadowColor": "#EEF2F5",
      "textColor": "#545456",
    }
  }
>
  <Link
    capitalize={false}
    italic={false}
    opacity={1}
    size="14px"
    uppercase={false}
    weight={500}
  >
    <styled.a
      capitalize={false}
      italic={false}
      opacity={1}
      size="14px"
      uppercase={false}
      weight={500}
    >
      <a
        className="c0"
        opacity={1}
        size="14px"
      >
        MyLink
      </a>
    </styled.a>
  </Link>
</ThemeProvider>
`;
